back/engines/free/email_campaigns/app/models/concerns/email_campaigns/trackable.rb

```diff
@@ -25,6 +25,10 @@ def last_delivery_for_recipient(user)
     end
 
     def extra_mailgun_variables
+      if !@delivery_id
+        ErrorReporter.report_msg('No delivery ID in extra_mailgun_variables!')
+        @delivery_id = SecureRandom.uuid
```

Comment:
If I can't figure out the proper solution, perhaps I could just generate the `delivery_id` here (using `||=`) and remove generate_id and perhaps undo the whole split for before_send and filter.